ui/components/app/simulation-details/simulation-details.stories.tsx

```diff
@@ -0,0 +1,154 @@
+      },
+      [ERC20_TOKEN_2_MOCK]: {
+        address: ERC20_TOKEN_2_MOCK,
+        symbol: 'USDC',
+        name: 'USD Coin',
+        iconUrl:
+          `https://static.metafi.codefi.network/api/v1/tokenIcons/1/${ERC20_TOKEN_2_MOCK}.png`,
+      },
+      [ERC721_TOKEN_MOCK]: {
+        address: ERC721_TOKEN_MOCK,
+        symbol: 'CK',
+        name: 'CryptoKitties',
+        iconUrl:
+          `https://static.metafi.codefi.network/api/v1/tokenIcons/1/${ERC721_TOKEN_MOCK}.png`,
+      },
+      [ERC1155_TOKEN_MOCK]: {
+        address: ERC1155_TOKEN_MOCK,
+        symbol: 'MAYC',
+        name: 'Mutant Ape Yacht Club',
+        iconUrl:
+          `https://static.metafi.codefi.network/api/v1/tokenIcons/1/${ERC1155_TOKEN_MOCK}.png `
+      }
+    },
+  },
+});
+
+const meta: Meta<typeof SimulationDetails> = {
+  title: 'Components/App/SimulationDetails',
+  component: SimulationDetails,
+  decorators: [(story) => <Provider store={storeMock}>{story()}</Provider>],
+};
+
+export default meta;
+type Story = StoryObj<typeof SimulationDetails>;
+
+export const BuyErc20: Story = {
+  args: {
+    simulationData: {
+      nativeBalanceChange: {
+        ...DUMMY_BALANCE_CHANGE,
+        difference: '0x12345678912345678',
+        isDecrease: true,
+      },
+      tokenBalanceChanges: [{
+        ...DUMMY_BALANCE_CHANGE,
+        address: ERC20_TOKEN_1_MOCK,
+        difference: '0x123456',
+        isDecrease: false,
+        standard: SimulationTokenStandard.erc20,
+      }, {
+        ...DUMMY_BALANCE_CHANGE,
+        address: ERC20_TOKEN_2_MOCK,
+        difference: '0x123456901',
+        isDecrease: false,
+        standard: SimulationTokenStandard.erc20,
+      }, {
+        ...DUMMY_BALANCE_CHANGE,
+        address: ERC721_TOKEN_MOCK,
+        difference: '0x1',
+        isDecrease: false,
+        id: '0x721',
+        standard: SimulationTokenStandard.erc721,
+      }, {
+        ...DUMMY_BALANCE_CHANGE,
+        address: ERC1155_TOKEN_MOCK,
+        difference: '0x13',
+        isDecrease: false,
+        id: '0x1155',
+        standard: SimulationTokenStandard.erc1155,
+      }],
+    },
+  },
+};
+
+
+export const PositiveNativeChange: Story = {
+  args: {
+    simulationData: {
+      nativeBalanceChange: {
+        previousBalance: '0x2',
+        newBalance: '0x1',
+        difference: '0x12345678912345678',
+        isDecrease: false,
+      },
+      tokenBalanceChanges: [],
+    },
+  },
+};
+
+export const NegativeNativeChange: Story = {
+  args: {
+    simulationData: {
+      nativeBalanceChange: {
+        previousBalance: '0x1',
+        newBalance: '0x2',
+        difference: '0x12345678912345678',
+        isDecrease: true,
+      },
+      tokenBalanceChanges: [],
+    },
+  },
+};
+
+export const NoBalanceChanges: Story = {
+  args: {
+    simulationData: {
+      nativeBalanceChange: undefined,
+      tokenBalanceChanges: [],
+    },
+  },
+};
+
+export const SimulationFailed: Story = {
```

Comment:
As per the TODOs, we're still pending descriptions for all these via comments.

Reply:
Hey Matt, Storybook automatically sets the export name as the label, and I've followed their [documentation examples](https://storybook.js.org/docs/writing-stories/args). The navigation looks very similar to the Name component:

![image](https://github.com/MetaMask/metamask-extension/assets/507015/8049f46a-459e-436e-bdec-e8dbcc563e20)

Is this what you were thinking, or is there something else you'd like me to tweak?